### PR TITLE
issue: 1027871 fix valgrind uninitialized value read errors

### PR DESCRIPTION
**Pull request by rafiw**
Signed-off-by: Rafi Wiener <rafiw@mellanox.com>

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1672/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1110/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1685/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1124/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/accl-libvma-pr/1686/ for details (Mellanox internal link).

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/accl-libvma-pr/1125/ for details (Mellanox internal link).

File: src/vma/dev/net_device_table_mgr.cpp [hunk 1]
```diff
@@ -726,7 +726,8 @@ void net_device_table_mgr::handle_timer_expired(void* user_data)
 void net_device_table_mgr::global_ring_wakeup()
 {
 	ndtm_logdbg("");
-	struct epoll_event ev;
+	epoll_event ev = {0, {0}};
```
Comment (ilansmith) [inline src/vma/dev/net_device_table_mgr.cpp]: Should do `{0}` and not `{0, {0}}`. It is more generic and a much more common and acceptable way of doing this.

Comment (rafiw) [inline src/vma/dev/net_device_table_mgr.cpp]: it doesn't compile

File: src/vma/util/wakeup.cpp [hunk 2]
```diff
@@ -55,9 +55,8 @@
 wakeup::wakeup()
 {
 	m_epfd = 0;
-        m_is_sleeping = 0;
-	m_ev.events = 0;
-	m_ev.data.fd = -1;
+	m_is_sleeping = 0;
+	memset(&m_ev, 0, sizeof(m_ev));
```
Comment (OphirMunk) [inline src/vma/util/wakeup.cpp]: @rafiw Here you are setting all m_ev to 0
```
memset(&m_ev, 0, sizeof(m_ev));
```
However previously fd was set to -1
```
m_ev.data.fd = -1;
```
Please explain

**Comment by ilansmith** [comment]
Merged #362

**Comment by ilansmith** [comment]
Merged #362

**Pull request closed without merging**